### ISSUE TITLE
Generate map style sheet in python to simplify string replacement jinja template approach

### ISSUE THREAD

**Issue by ryanbaumann** [mapbox/mapboxgl-jupyter#23]
# Background
Currently, there is business logic occurring in both the Python `mapboxgl` library and the javascript templates.  Since this is a Python library, we should isolate all business logic to Python, and make the Javascript component a true template only.

# Approach

1. Generate a complete [Mapbox GL Style Sheet](https://www.mapbox.com/mapbox-gl-js/style-spec/) in a Python dictionary in the [viz.py](https://github.com/mapbox/mapboxgl-jupyter/blob/master/mapboxgl/viz.py)
2. Serve Mapbox GL Style to the javascript template for visual.
3. Each time the analysis updates, the templates do not need to be destroyed & recreated - they simply run map.setStyle(newStyleURL)

cc @perrygeo

**Comment by ryanbaumann** [comment]
This proposal would solve https://github.com/mapbox/mapboxgl-jupyter/issues/19

**Comment by murphy214** [comment]
This could be done with a websocket as well! Have you ever considered using something with a little more structure like a protocol buffer thats serialized as [json](https://developers.google.com/protocol-buffers/docs/proto3#json) if you knew what your were doing you could probably scrape the page with the style sheet spec on it to generate the protocol buffer structures based on the required fields etc. I'm not saying it'd be easy, in fact it'd probably be a pain but once you did once you could probably do it as things are updated (assuming that style-spec page stays up to date.) lol I may have to try this now!

**Comment by ryanbaumann** [comment]
Working towards this in https://github.com/mapbox/mapboxgl-jupyter/pull/48

**Comment by akacarlyann** [comment]
This is going to help so much with multi-layer maps if we can include the layer definitions!!  Been playing around with serializing the property expressions on the Python side to get styled heatmaps layered under a point and linestring viz.  Fun stuff : )

<img width="821" alt="screen shot 2018-04-18 at 10 53 54 pm" src="https://user-images.githubusercontent.com/13527707/38973659-ce206a46-435b-11e8-9144-10bc44429bc1.png">

**Comment by ryanbaumann** [comment]
@akacarlyann yeah this will unblock a lot of use cases - and it looks like you have a POC already!  Is it something you're planning to PR or just an experiment?

**Comment by ryanbaumann** [comment]
This is a blocker for https://github.com/mapbox/mapboxgl-jupyter/issues/34
